src/@utils/order.ts

```diff
@@ -155,18 +156,20 @@ async function approveProviderFee(
   web3: Web3,
   providerFeeAmount: string
 ): Promise<string> {
+  const config = getOceanConfig(asset.chainId)
   const baseToken =
     asset?.accessDetails?.type === 'free'
       ? getOceanConfig(asset.chainId).oceanTokenAddress
       : asset?.accessDetails?.baseToken?.address
   const txApproveWei = await approveWei(
     web3,
+    config,
     accountId,
     baseToken,
     asset?.accessDetails?.datatoken?.address,
     providerFeeAmount
   )
-  return txApproveWei as string // thanks ocean.js
+  return txApproveWei as unknown as string // thanks ocean.js
```

Comment:
i will double check this on ocean.js to see if we can get rid of those casts, does not look really good